src/login/logind-session-device.c

```diff
@@ -367,9 +359,39 @@ int session_device_new(Session *s, dev_t dev, SessionDevice **out) {
         r = hashmap_put(s->devices, &sd->dev, sd);
         if (r < 0) {
                 r = -ENOMEM;
+                free(sd->node);
```

Comment:
Agreed it looks strange. But this was already the case, the commit only moved the code around. I think we could revisit this later.